mule-deployer/src/main/java/org/mule/tools/model/anypoint/ArmDeployment.java

```diff
@@ -23,39 +26,98 @@
  * LICENSE.txt file.
  */
 
-public class ArmDeployment extends DeploymentConfiguration implements AnypointDeployment {
+public class ArmDeployment implements AnypointDeployment {
 
-  @Parameter(required = false, property = "anypoint.username")
+  @Parameter
   protected String username;
 
-  @Parameter(required = false, property = "anypoint.password")
+  @Parameter
   protected String password;
 
-  @Parameter(required = false, property = "anypoint.environment")
+  @Parameter
   protected String environment;
 
-  @Parameter(defaultValue = "", property = "anypoint.businessGroup")
-  protected String businessGroup = "";
+  @Parameter
+  protected String businessGroup;
 
-  @Parameter(readonly = true, property = "anypoint.uri", defaultValue = "https://anypoint.mulesoft.com")
+  @Parameter
   protected String uri;
 
-  @Parameter(property = "anypoint.target")
+  @Parameter
   protected String target;
 
-  @Parameter(property = "anypoint.target.type")
+  @Parameter
   protected TargetType targetType;
 
-  @Parameter(defaultValue = "Medium", readonly = true, property = "arm.insecure")
-  protected boolean armInsecure;
+  @Parameter
+  protected Boolean armInsecure;
 
-  @Parameter(defaultValue = "true")
-  protected boolean failIfNotExists;
+  @Parameter
+  protected Boolean failIfNotExists;
 
   // TODO validate what for?
-  @Parameter(required = false, property = "maven.server")
+  @Parameter
   protected String server;
 
+  // TODO change name to artifact
+  @Parameter
+  protected File application; // VALIDATIONS REQURIED
```

Comment:
shouldn't we call this artifact instead of application?

Reply:
done